tests/components/technove/test_sensor.py

```diff
@@ -0,0 +1,161 @@
+
+    assert (state := hass.states.get("sensor.technove_station_free_memory"))
+    assert state.attributes.get(ATTR_ICON) == "mdi:memory"
+    assert state.attributes.get(ATTR_UNIT_OF_MEASUREMENT) == UnitOfInformation.BYTES
+    assert state.state == "14600"
+    assert entry.entity_category is EntityCategory.DIAGNOSTIC
+
+    assert (entry := entity_registry.async_get("sensor.technove_station_free_memory"))
+    assert entry.unique_id == "AA:AA:AA:AA:AA:BB_free_heap"
+    assert entry.entity_category is EntityCategory.DIAGNOSTIC
+
+    assert (state := hass.states.get("sensor.technove_station_wi_fi_signal"))
+    assert state.attributes.get(ATTR_ICON) == "mdi:wifi"
+    assert state.attributes.get(ATTR_UNIT_OF_MEASUREMENT) == PERCENTAGE
+    assert state.state == "76"
+    assert entry.entity_category is EntityCategory.DIAGNOSTIC
+
+    assert (entry := entity_registry.async_get("sensor.technove_station_wi_fi_signal"))
+    assert entry.unique_id == "AA:AA:AA:AA:AA:BB_wifi_signal"
+    assert entry.entity_category is EntityCategory.DIAGNOSTIC
+
+    assert (state := hass.states.get("sensor.technove_station_wi_fi_rssi"))
+    assert state.attributes.get(ATTR_DEVICE_CLASS) == SensorDeviceClass.SIGNAL_STRENGTH
+    assert (
+        state.attributes.get(ATTR_UNIT_OF_MEASUREMENT)
+        == SIGNAL_STRENGTH_DECIBELS_MILLIWATT
+    )
+    assert state.state == "-62"
+
+    assert (entry := entity_registry.async_get("sensor.technove_station_wi_fi_rssi"))
+    assert entry.unique_id == "AA:AA:AA:AA:AA:BB_wifi_rssi"
+    assert entry.entity_category is EntityCategory.DIAGNOSTIC
+
+    assert (state := hass.states.get("sensor.technove_station_wi_fi_channel"))
+    assert state.attributes.get(ATTR_ICON) == "mdi:wifi"
+    assert state.attributes.get(ATTR_UNIT_OF_MEASUREMENT) is None
+    assert state.state == "11"
+
+    assert (entry := entity_registry.async_get("sensor.technove_station_wi_fi_channel"))
+    assert entry.unique_id == "AA:AA:AA:AA:AA:BB_wifi_channel"
+    assert entry.entity_category is EntityCategory.DIAGNOSTIC
+
+    assert (state := hass.states.get("sensor.technove_station_wi_fi_bssid"))
+    assert state.attributes.get(ATTR_ICON) == "mdi:wifi"
+    assert state.attributes.get(ATTR_UNIT_OF_MEASUREMENT) is None
+    assert state.state == "AA:AA:AA:AA:AA:BB"
+
+    assert (entry := entity_registry.async_get("sensor.technove_station_wi_fi_bssid"))
+    assert entry.unique_id == "AA:AA:AA:AA:AA:BB_wifi_bssid"
+    assert entry.entity_category is EntityCategory.DIAGNOSTIC
+
+    assert (state := hass.states.get("sensor.technove_station_ip"))
+    assert state.attributes.get(ATTR_ICON) == "mdi:ip-network"
+    assert state.attributes.get(ATTR_UNIT_OF_MEASUREMENT) is None
+    assert state.state == "127.0.0.1"
+
+    assert (entry := entity_registry.async_get("sensor.technove_station_ip"))
+    assert entry.unique_id == "AA:AA:AA:AA:AA:BB_ip"
+    assert entry.entity_category is EntityCategory.DIAGNOSTIC
```

Comment:
You'll love snapshot testing. Checkout https://github.com/home-assistant/core/blob/59a01fcf9c42a233ecaa92c6328116128569881c/tests/components/withings/test_sensor.py#L26-L46

If you use SnapshotAssertion in your test, execute your tests with `pytest ./tests/components/technove --snapshot-update` and it will generate .ambr files which capture your objects

Reply:
Yes, that's much nicer! Done.